test_util/cluster.py

```diff
@@ -0,0 +1,260 @@
+import itertools
+import logging
+import os
+import random
+import stat
+import string
+from collections import namedtuple
+from contextlib import closing, contextmanager
+from os.path import join
+
+import passlib.hash
+import pkg_resources
+from retrying import retry
+
+import test_util.ccm
+import test_util.installer_api_test
+import test_util.test_runner
+from ssh.ssh_tunnel import SSHTunnel, TunnelCollection
+
+
+Host = namedtuple('Host', ['public_ip', 'private_ip'])
+
+
+class TestCluster:
+
+    def __init__(
```

Comment:
This interface seems very strongly bound / non-generalizable beyond the "CCM VPC" type clusters, would be nice for it to work on any list of Host() objects provided, with a Vpc object which generalizes the "delete" type interface.